pulser-core/pulser/channels/eom.py

```diff
@@ -250,16 +330,26 @@ def _lightshift(
     def _rabi_freq_per_beam(
         self, rabi_frequency: float
     ) -> dict[RydbergBeam, float]:
+        shift_factor = np.sqrt(
+            self.red_shift_coeff / self.blue_shift_coeff
+            if self.limiting_beam == RydbergBeam.RED
+            else self.blue_shift_coeff / self.red_shift_coeff
+        )
         # rabi_freq = (rabi_red * rabi_blue) / (2 * int_detuning)
-        limit_rabi_freq = self.max_limiting_amp**2 / (
-            2 * self.intermediate_detuning
+        limit_rabi_freq = (
+            shift_factor
+            * self.max_limiting_amp**2
+            / (2 * self.intermediate_detuning)
         )
         # limit_rabi_freq is the maximum effective rabi frequency value
-        # below which the rabi frequency of both beams can be matched
+        # below which the lightshift can be zero
         if rabi_frequency <= limit_rabi_freq:
-            # Both beams the same rabi_freq
-            beam_amp = np.sqrt(2 * rabi_frequency * self.intermediate_detuning)
-            return {beam: beam_amp for beam in RydbergBeam}
+            base_amp = np.sqrt(2 * rabi_frequency * self.intermediate_detuning)
+            sqrt_shift_factor = np.sqrt(shift_factor)
+            return {
+                self.limiting_beam: base_amp / sqrt_shift_factor,
+                ~self.limiting_beam: base_amp * sqrt_shift_factor,
+            }
```

Comment:
```suggestion
            base_amp_squared = 2 * rabi_frequency * self.intermediate_detuning
            return {
                self.limiting_beam: np.sqrt(base_amp / shift_factor),
                ~self.limiting_beam: np.sqrt(base_amp * shift_factor),
            }
```
or surely even better
```suggestion
            base_amp_lim = np.sqrt(2 * rabi_frequency * self.intermediate_detuning / shift_factor)
            return {
                self.limiting_beam: base_amp_lim,
                ~self.limiting_beam: base_amp_lim * shift_factor,
            }
```

Reply:
The first suggestion looks better to me, though it's missing the `_squared` in the last two `base_amp`s
```suggestion
            base_amp_squared = 2 * rabi_frequency * self.intermediate_detuning
            return {
                self.limiting_beam: np.sqrt(base_amp_squared / shift_factor),
                ~self.limiting_beam: np.sqrt(base_amp_squared * shift_factor),
            }
```

Reply:
Oops indeed ;) I think second one is "faster" but eventually less clear and we are definitely not optimising down to the number of sqrt ^^'